### PR TITLE
5322: Add post request

### PR DESCRIPTION
**Pull request by kat1906**
Story: [AB#5322](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/5322)
Task: [AB#6777](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6777)

### CODE REVIEW

File: app/pages/items/description/contextCreator.js [hunk 1]
```diff
@@ -4,5 +4,4 @@ import { baseUrl } from '../../../config';
 export const getContext = ({ orderId }) => ({
```
Comment (alibombaye) [inline app/pages/items/description/contextCreator.js:4]: 🙄  can we change the folder name from `items` to `sections` please

Comment (kat1906) [reply app/pages/items/description/contextCreator.js:4]: nooooooooooooooooooooooooooo........

File: app/pages/items/description/controller.js [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+  try {
+    // TODO: call putData if not new order
+    const response = isNewOrder ? await postData(apiCallParams) : {};
+    logger.info(`Order ${isNewOrder ? 'added' : 'updated'} - id: ${response.data.id}, ${JSON.stringify(body)}`);
+    return { success: true, orderId: isNewOrder ? response.data.orderId : orderId };
+  } catch (err) {
+    if (err.response.status === 400 && err.response.data && err.response.data.errors) {
+      // TODO: validation
+      return { success: false };
+    }
```
Comment (alibombaye) [inline app/pages/items/description/controller.js:36]: 👍 looks good

File: app/pages/items/description/ui.test.js [hunk 3]
```diff
@@ -105,22 +106,25 @@ test('should render save button', async (t) => {
...
-  const button = Selector('[data-test-id="save-button"] a');
+  const saveButton = Selector('[data-test-id="save-button"] button');
 
   await t
-    .expect(button.exists).ok()
-    .click(button)
-    .expect(getLocation()).eql('http://localhost:1234/organisation/neworder/description#');
+    .expect(saveButton.exists).ok()
+    .click(saveButton)
+    .expect(getLocation()).eql('http://localhost:1234/order/organisation/order1');
```
Comment (alibombaye) [inline app/pages/items/description/ui.test.js:129]: I've been using the baseUrl for the `/order`. but happy to go with either way. I actually prefer it the way you have done it. Long as we stick to the one way

File: app/routes.test.js [hunk 4]
```diff
@@ -37,6 +42,55 @@ const checkAuthorisedRouteNotLoggedIn = path => (
...
+    .post(postPath)
+    .type('form')
+    .set('Cookie', [cookies, mockUnauthorisedCookie])
+    .send({ _csrf: csrfToken })
+    .expect(200)
+    .then((res) => {
+      expect(res.text.includes('data-test-id="error-title"')).toEqual(true);
+      expect(res.text.includes('You are not authorised to view this page')).toEqual(true);
+    });
+};
```
Comment (alibombaye) [inline app/routes.test.js:92]: we really need to move the above 3 into BCL. I'll add to tech-debt.

File: app/pages/items/description/ui.test.js (hunk 3 above)
Comment (kat1906) [reply app/pages/items/description/ui.test.js:129]: I did have that but then I took it out 🤷 doesn't make any difference as it is the same for all envs

File: app/routes.test.js (hunk 4 above)
Comment (kat1906) [reply app/routes.test.js:92]: at this point is it worth it?

Comment (alibombaye) [reply app/routes.test.js:92]: I've added the ticket. If we get round to it then it'll be nice. It just clutters up the routes.test.js with about 30 lines before you get to the main bit

**Pull request merged**